packages/react-core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -39,6 +39,8 @@ export interface DatePickerProps
   popoverProps?: Omit<PopoverProps, 'appendTo'>;
   /** Functions that returns an error message if a date is invalid */
   validators?: ((date: Date) => string)[];
+  /** Additional props for input field */
+  inputProps?: object;
```

Comment:
Try `TextInputProps`. You'll need to add `import { TextInputProps } from '../TextInput';` to the top of the file.